### PR TITLE
Enhance/display important dates on search

### LINKED ISSUE #41075: Display only most important date when listing posts or pages
We currently display 2 dates on wordpress.com/posts/scheduled and wordpress.com/pages/scheduled

Screen Shot 2020-04-14 at 11 22 10

First one refers to the last modified date while the latter indicates when the post will be published. 

In order to reduce the amount of copy, let's remove the last modified date entirely and only display the date t's scheduled.

That way, we'll always display one date depending on post status:
- Published date for published posts.
- Scheduled date for scheduled posts.
- Last modified date for draft posts.
- Trashed date for trashed posts.

To make clear what the date means, we'd need to add appropriate labels in a similar fashion to WP Admin:

Screen Shot 2020-04-07 at 2 27 34 PM

### PR DESCRIPTION
**Pull request by millerf**
#### Changes proposed in this Pull Request

* Depending on post/page status, display the most meaningful date
* On search, display the said date with the status label
* ~Small~ Good Refactor of `PostRelativeTime`

 Search results should look close to this mockup:
Screenshot 2020-04-15 at 00 31 49

#### Testing instructions

* Go to pages or posts views. Make sure displayed dates are the right ones
* When searching, make sure the dates are the same and label is displayed

#### Extras
Found a bug reported here
https://github.com/Automattic/wp-calypso/issues/41245
As it is out of scope for this one.

Fixes #41075

cc @mmtr @sixhours

### CODE REVIEW

File: client/my-sites/post-relative-time-status/index.jsx [hunk 1]
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
...
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
-		}
-
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		const scheduledTime = scheduledDate.calendar( null, {
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:64]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41242)

File: client/my-sites/post-relative-time-status/index.jsx [hunk 2]
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
...
-		} else if ( this.props.includeBasicStatus ) {
-			if ( status === 'draft' ) {
-				statusText = this.props.translate( 'draft' );
-			} else if ( status === 'publish' ) {
-				statusText = this.props.translate( 'published' );
-			} else if ( status === 'new' ) {
-				statusText = this.props.translate( 'Publish immediately' );
-			}
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayScheduleTime)s', {
```
Comment (a8ci18n) [inline client/my-sites/post-relative-time-status/index.jsx:124]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'scheduled for %(displayScheduleTime)s' )` *ES Score: 8*

**Comment by millerf** [comment]
`PostRelativeTime` was refactored.

Posts and Pages views were modified according to specs:


| Before | After | Comment|
| ---- | ---- | ----|
| Screenshot 2020-04-18 at 19 21 32 | Screenshot 2020-04-18 at 19 21 17 |  On listing, scheduled posts/pages now display the right date (scheduled ate) |
| Screenshot 2020-04-18 at 19 23 56 |  Screenshot 2020-04-18 at 19 25 06 | Published now display "sticky" if needed | 
| Screenshot 2020-04-18 at 19 29 19  |  Screenshot 2020-04-18 at 19 26 25  | Post Search now displays the right status and appropriate date. The "sticky" status has been added. Maybe need some feedback from design? |
|  Screenshot 2020-04-18 at 19 29 05 | Screenshot 2020-04-18 at 19 31 03| Pages search now displays only one date and the right status |

File: client/my-sites/post-relative-time-status/index.jsx (hunk 1 above)
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:64]: `'LL [at] LT' ` has been modified and is now ` '[on] ll [at] LT'`

File: client/my-sites/post-relative-time-status/index.jsx (hunk 2 above)
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:124]: `'scheduled for %(displayScheduleTime)s'` has been modified and is now  `'scheduled %(displayScheduleTime)s'`

**Comment by sixhours** [comment]
Thanks @millerf ! It looks like this needs to be rebased, it's asking me to use `npm` when we recently switched to `yarn`. Give that a try and let us know if you need help with it!

**Review by sixhours — COMMENTED** [review]
This is not a comprehensive review. Please disregard if this is still "in progress"!

I noticed a couple issues when I fired this up:

When searching posts, the date shows "Published modified on [date]":
Screen Shot 2020-04-20 at 11 48 25 AM

When searching for a scheduled post, the line "Scheduled for Tomorrow at [time]" reads oddly because "tomorrow" is capitalized:
Screen Shot 2020-04-20 at 12 25 31 PM

The latter issue isn't unique to this PR (I noticed the same behavior on prod) but might be something we could address. Totally OK if that's out of scope and we address in a different PR: let me know and I'll create an issue.

File: client/my-sites/pages/page-card-info/index.jsx [hunk 3]
```diff
@@ -62,7 +62,7 @@ function PageCardInfo( {
 						post={ page }
 						link={ contentLink.contentLinkURL }
 						target={ contentLink.contentLinkTarget }
-						gridiconSize={ ICON_SIZE }
+						gridIconSize={ ICON_SIZE }
```
[the remark this answers is not in the document]
Comment (millerf) [reply client/my-sites/pages/page-card-info/index.jsx:65]: I'll revert the commit

File: client/blocks/post-item/index.jsx [hunk 4]
```diff
@@ -211,7 +211,7 @@ class PostItem extends React.Component {
 										post={ post }
 										link={ enabledPostLink }
 										target={ null }
-										gridiconSize={ ICON_SIZE }
+										gridIconSize={ ICON_SIZE }
```
[the remark this answers is not in the document]
Comment (millerf) [reply client/blocks/post-item/index.jsx:214]: Damn... I tried to look smart... 😄

**Comment by millerf** [comment]
> "Published modified on [date]"

I see. What would be the correct way to say it... the idea is that the status of the post is published but the date is the last modified date.
Would "Published, last modified on [date]" be acceptable?

> "Scheduled for Tomorrow at [time]" 

It is definitely in the scope. I'll fix it.

I am having some trouble to push since I rebased. I don't think it is linked to the rebase, as I seem to have the same trouble on new branches...
```
refusing to allow an OAuth App to create or update workflow `.github/workflows/full-site-editing-plugin.yml` without `workflow` scope
``` 
So not sure if I would be able to address all that tonight.

**Comment by sixhours** [comment]
> Would "Published, last modified on [date]" be acceptable?

Yes, that sounds right to me!

> It is definitely in the scope. I'll fix it.

Yay! Thank you!

> I am having some trouble to push since I rebased. I don't think it is linked to the rebase, as I seem to have the same trouble on new branches...

That's a new one to me. 😬 It sounds like something weird with Git/Github? Hoping @kwight can offer some insight.

**Comment by millerf** [comment]
> It sounds like something weird with Git/Github? Hoping @kwight can offer some insight.

It is bizarre as I am pushing on my fork... Also using command-line. All help I found online relate to git GUIs... I might just scratch the local repo...

**Comment by sixhours** [comment]
> I might just scratch the local repo...

That's my go-to for fixing weirdness, too. Burn it all to the ground. :D

**Comment by millerf** [comment]
> > I might just scratch the local repo...
> 
> That's my go-to for fixing weirdness, too. Burn it all to the ground. :D

Wow... I deleted the local repo, updated git, nothing worked. I ended up downloading Github Desktop and it worked straight away... Something about OAuth tokens that weren't saved the right way or something. Github desktop manages it in the automatically in the background and it worked...

File: client/my-sites/post-relative-time-status/index.jsx [hunk 5]
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
 	static propTypes = {
 		showPublishedStatus: PropTypes.bool.isRequired,
 		post: PropTypes.object.isRequired,
-		includeNonDraftStatuses: PropTypes.bool,
 		link: PropTypes.string,
 		target: PropTypes.string,
-		gridiconSize: PropTypes.number,
+		gridIconSize: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (millerf) [reply client/my-sites/post-relative-time-status/index.jsx:25]: resolved on #41301

**Comment by millerf** [comment]
Work continues on #41301

**Pull request closed without merging**